erpnext/assets/doctype/asset_maintenance/asset_maintenance.py

```diff
@@ -16,26 +16,44 @@ def validate(self):
 				throw(_("Start date should be less than end date for task {0}").format(task.maintenance_task))
 			if getdate(task.next_due_date) < getdate(nowdate()):
 				task.maintenance_status = "Overdue"
-			if not self.get("__islocal"):
-				if not task.assign_to:
-					task.assign_to = self.maintenance_manager
-				if task.assign_to:
-					self.assign_tasks(task)
+
+	def save(self):
```

Comment:
Why you overwritten `save` method of document.py? You can call `sync_maintenance_tasks` method in `after_insert` or `on_update`.